Extensions/Share/ShareViewController.swift

```diff
@@ -0,0 +1,306 @@
+
+        [label, checkmark].forEach {
+            blue.addSubview($0)
+            $0.textColor = .white
+        }
+
+        label.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.left.equalToSuperview().inset(UX.pageInfoRowLeftInset)
+            make.right.equalTo(checkmark.snp.left)
+        }
+
+        checkmark.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.right.equalToSuperview().inset(UX.rowInset)
+            make.width.equalTo(20)
+        }
+    }
+
+    private func setupNavBar() {
+        navigationController?.navigationBar.isTranslucent = false
+        navigationController?.navigationBar.setValue(true, forKey: "hidesShadow") // hide separator line
+        navigationItem.titleView = UIImageView(image: UIImage(named: "Icon-Small"))
+        navigationItem.titleView?.contentMode = .scaleAspectFit
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: Strings.SendToCancelButton, style: .plain, target: self, action: #selector(finish))
+    }
+
+    private func setupStackView() {
+        stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.distribution = .fill
+        stackView.alignment = .fill
+        stackView.spacing = 4
+        view.addSubview(stackView)
+        stackView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = .white
+
+        setupNavBar()
+        setupStackView()
+
+        let (currentPageInfoRow, pageTitleLabel, urlLabel) = makePageInfoRow()
+
+        let trailingSpace = UIView()
+
+        let rows = [
+            currentPageInfoRow,
+            makeSeparator(),
+            makeActionRow(label: Strings.ShareLoadInBackground, imageName: "menu-Show-Tabs", action: actionLoadInBackground, hasNavigation: false),
+            makeActionRow(label: Strings.ShareBookmarkThisPage, imageName: "AddToBookmarks", action: actionBookmarkThisPage, hasNavigation: false),
+            makeActionRow(label: Strings.ShareAddToReadingList, imageName: "AddToReadingList", action: actionAddToReadingList, hasNavigation: false),
+            makeSeparator(),
+            makeActionRow(label: Strings.ShareSendToDevice, imageName: "menu-Send-to-Device", action: actionSendToDevice, hasNavigation: true),
+            trailingSpace
+        ]
+
+        rows.forEach {
+            stackView.addArrangedSubview($0)
+        }
+
+        // Without some growable space at the bottom there are constraint errors because the UIView space doesn't subdivide equally, and none of the rows are growable
+        trailingSpace.snp.makeConstraints { make in
+            make.height.greaterThanOrEqualTo(1)
+        }
+
+        layoutSeparators()
+
+        actionRows.forEach {
+            $0.snp.makeConstraints { make in
```

Comment:
I know it might be a big change now. But using stackviews all the way down will simplify how many constraints you need to specify. Especially because a lot of these constraints are fixed numbers. Constants might not work well in LTR or with l10n if the text needs to extend to 2 lines.

Reply:
interesting, let me try quickly and see if it looks like a big change.